### DISCUSSION TITLE
Hooks in standalone mode?

### DISCUSSION THREAD

**Discussion by ooralas** [webdriverio/webdriverio#10140] [Ideas]
Hello 👋, 
is it possible to use hooks in standalone mode (i.e. i am not testing anything, i want to build an automation framework to automating some tasks) ? i am discovering this great framework and i am wondering if this is possible.

**Comment by christian-bromann** [comment]
Hey @ooralas 👋
what hooks are you interested in? If you use WebdriverIO as [standalone](https://webdriver.io/docs/setuptypes) tool for automation purposes you usually should not need these hooks as you have all the chances to call async function at any point in time (e.g. before automation process starts or after etc.).

**Reply by ooralas** [reply]
Hi @christian-bromann 👋 thanks for your reply.
as i said i am trying the library and want to try a few things 😃.
So far it all worked for me. But sometimes the documentation is missing info for things I want to use in standalone mode for example: the hooks, because the documentation always talks about the Testrunner, so I wanted to know if this is possible, and how it possible ?

So in the future you will see my messages in the disscutions and the slack channel. 😉

**Reply by christian-bromann** [reply]
Hey @ooralas yeah, welcome to the WebdriverIO community 🙋 Feel free to shoot any questions in the channels the project provides.

Note that WebdriverIO is mainly used for testing purposes but can be used for everything related to browser or remote device automation as well. You can also use the testrunner for your automation process even though you don't test anything. See it as a execution construct.

> So far it all worked for me. But sometimes the documentation is missing info for things I want to use in standalone mode for example: the hooks

Can you be more specific as to where the docs are confusing or where documentation is missing?

**Reply by ooralas** [reply]
Hi @christian-bromann 😃 ,

> You can also use the testrunner for your automation process even though you don't test anything. See it as a execution construct.

sounds very interesting. Do you have any source or example project that I can follow ? 

> Can you be more specific as to where the docs are confusing or where documentation is missing?

Sorry ich habe glaube ich falsch ausgedrückt. Die Dokumentation von Webdriverio sind wirklich toll und sehr informativ. Ich meinte aber, dass bei manchen Modulen die Erklärung bezieht sich hauptsächlich um Testrunner und fehlen Informationen, wie man sie in der Standalone-mode nutzen kann, wie z.B.: selenium-standalone-service.

**Reply by christian-bromann** [reply]
> Do you have any source or example project that I can follow ?

see for example our Jasmine boilerplate: https://github.com/webdriverio/jasmine-boilerplate

> wie man sie in der Standalone-mode nutzen kann, wie z.B.: selenium-standalone-service.

You can't use these services in standalone mode.

**Reply by ooralas** [reply]
Sorry, I wrote the first part of my message in English and the rest in German. That can sometimes happen when you have both the Work's Slack app and Browser open at the same time 😅.

**Reply by christian-bromann** [reply]
no problem!

**Reply by ooralas** [reply]
> You can't use these services in standalone mode.

@christian-bromann  Does that mean I can not use all the services one sees in the documentation for automation (not test automation)?

**Reply by christian-bromann** [reply]
you can not use them outside the testrunner, however you can also just use the testrunner for any type of automation.

**Reply by ooralas** [reply]
@christian-bromann  okay. 
I have already looked at the example you suggested. I don't know if this is suitable for my use case. 
My automation scripts will live on a RESTFull API, depending on the endpoint different scripts will be executed. how is it possible to do something like this with the testrunner? 🤔

**Reply by christian-bromann** [reply]
You can also always use services within standalone mode by calling the class hooks yourself at the right time.
